src/lambda/rest.eventmgmtapi.getEventParticipant.ts

```diff
@@ -5,20 +5,19 @@ import { operations } from '../generated/rest.eventmgmtapi-model.generated';
 export const handler = api.createOpenApiHandler<operations['getEventParticipant']>(async (ctx) => {
   ctx.logger.info(JSON.stringify(ctx.event));
 
-  const participantId = decodeURIComponent(ctx.event.pathParameters!.participantID!); // id is an e-mail address and thus encoded
   const eventId = ctx.event.pathParameters!.eventID;
   const token = ctx.event.queryStringParameters!.token;
 
-  const participant = await Participant.get({ eventId: eventId, email: participantId });
-  if (!participant || (token !== participant.token && !ctx.auth.isAdmin())) {
+  const participant = await Participant.get({ eventId: eventId, token: token });
+  if (token !== participant?.token && !ctx.auth.isAdmin()) {
```

Comment:
!participant sollte auch NotFound werfen, dann muss auch das ? unten nicht sein.

Reply:
re-added the condition, also changed that in the other functions, where applicable.